OpenRA.Mods.Common/Traits/Attack/AttackTurreted.cs

```diff
@@ -35,11 +37,13 @@ protected override bool CanAttack(Actor self, Target target)
 			if (!base.CanAttack(self, target))
 				return false;
 
+			// Rotate all turrets to make sure upgrading a turret does not change the turret's facing.
+			var faceTarget = false;
 			foreach (var t in turrets)
-				if (t.FaceTarget(self, target))
-					return true;
+				if (t.FaceTarget(self, target) && !t.IsTraitDisabled)
```

Comment:
Be aware of `FaceTarget`'s side effects. Do you want the turret to turn towards the target even when that turret is disabled? Otherwise the `!t.IsTraitDisabled` needs to be the first check in the condition.